### ISSUE TITLE
fast_float segfault

### ISSUE THREAD

**Issue by butcherg** [mm2/Little-CMS#236]
Encountered a read access segfault transforming an image for display in rawproc, my hack software.  here's the address sanitizer report:

```
==195138==ERROR: AddressSanitizer: SEGV on unknown address 0x7feccf01e840 (pc 0x5560cf797bb7 bp 0x000000000000 sp 0x7feed1c7b4c0 T22)
==195138==The signal is caused by a READ memory access.
    #0 0x5560cf797bb7 in flerp ../../../../plugins/fast_float/src/fast_float_internal.h:142
    #1 0x5560cf797bb7 in flerp ../../../../plugins/fast_float/src/fast_float_internal.h:117
    #2 0x5560cf797bb7 in MatShaperFloat ../../../../plugins/fast_float/src/fast_float_matsh.c:193
    #3 0x5560cf77a859 in cmsDoTransformLineStride ../../src/cmsxform.c:248
    #4 0x5560cee68fe5 in gImage::getImageDataForDisplay(bool, void*, void*, unsigned int, unsigned int) [clone ._omp_fn.2] ../../src/gimage.cpp:616
    #5 0x7feef58b577d  (/lib/x86_64-linux-gnu/libgomp.so.1+0x1a77d)
    #6 0x7feef588258f in start_thread nptl/pthread_create.c:463
    #7 0x7feef578b222 in __clone (/lib/x86_64-linux-gnu/libc.so.6+0x119222)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV ../../../../plugins/fast_float/src/fast_float_internal.h:142 in flerp
Thread T22 created by T0 here:
    #0 0x5560ced4b795 in __interceptor_pthread_create (/home/glenn/ImageStuff/rawproc10/rawproc/build-linux-sanitize/src/rawproc+0x314795)
    #1 0x7feef58b5dda  (/lib/x86_64-linux-gnu/libgomp.so.1+0x1adda)
    #2 0x7feef58ad8e0 in GOMP_parallel (/lib/x86_64-linux-gnu/libgomp.so.1+0x128e0)
    #3 0x7feeb76f67ff  (<unknown module>)

==195138==ABORTING
```
This happened transforming the image produced at a subtract tool, which in this case produced a lot of negative values in the image...

Edit: I take that back, just inspected my code, and I'm doing a fmin(fmax(img[i].*,0.0),1.0) in the assembly of the image array passed into cmsDoTransformLineStride().

**Comment by butcherg** [comment]
Cripes, nevermind.  Turns out I'm sending NaNs...

**Comment by mm2** [comment]
Thanks for let me know,

Anyway make sure you are using latest GitHub sources, or at least you have  fix 75bd87ec8d2bca9dfc766e8067c22a3d72e968a2

On not-a-numbers, a sanitisation would be terribly expensive in terms of throughput. I bet the fix above will solve the issue.

Regards
Marti

**Comment by butcherg** [comment]
Cleaned up and guarded my tone line math,slope calculation, all good now.
